gateway/utils/validate.go

```diff
@@ -186,9 +186,28 @@ func Validate(where map[string]interface{}, obj interface{}) bool {
 						return false
 					}
 				case "$contains":
-					switch val.(type) {
+					switch v := v2.(type) {
 					case map[string]interface{}:
-						return true
+						// check if result contains specified contains field
+						result, ok := res[k]
+						if !ok {
+							return false
+						}
+						resultObj, ok := result.(map[string]interface{})
+						if !ok {
+							return false
+						}
+						// check if the top level key values are same
+						for key, value := range v {
```

Comment:
This logic is incorrect. You need to check if all top level fields are equal. Correctly you are returning true even if one matches

Reply:
that's how it works in postgres. that' why i wrote the logic like that.

Reply:
Postgres matches a doc even if one key matches? And what about nested fields?

Reply:
yes, same applies for nested fields

Reply:
![image](https://user-images.githubusercontent.com/24411676/76676317-3d17a800-65e8-11ea-97d6-2168686ef519.png)

Reply:
do i use recursion for checking the nested fields ?
or is just enough to compare to interface using "=="

Reply:
Use recursion